### PR TITLE
Fallback to read properties file from classloader of the class

### LINKED ISSUE #1285: Warning log when oshi.properties not found
Hi, 

Thanks for the useful library
We have a warning log `No default configuration found` when loading oshi where the `Thread.currentThread().getContextClassLoader()` is not the classloader that loads the oshi jar 

https://github.com/oshi/oshi/blob/1e56bf1be27f44fdedcfb98dc65f95443cccb5da/oshi-core/src/main/java/oshi/util/FileUtil.java#L226

I have a proposal that if the resources is empty from the given classloader, we could try the ClassLoader of GlobalConfig.class.getClassLoader()

do you think it make sense?
I can create the pull request if it looks good to you

### PR DESCRIPTION
**Pull request by ngyukman**
resolve #1285

### CODE REVIEW

File: oshi-core/src/main/java/oshi/util/FileUtil.java [hunk 1]
```diff
@@ -217,40 +221,56 @@ public static String getStringFromFile(String filename) {
...
-        // Load the configuration file from the classpath
-        try {
-            ClassLoader loader = Thread.currentThread().getContextClassLoader();
-            if (loader == null) {
-                loader = ClassLoader.getSystemClassLoader();
-                if (loader == null) {
-                    throw new IOException();
-                }
+        // Load the configuration file from the different classloaders
+        Set<ClassLoader> loaders = Stream.of(
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/util/FileUtil.java:234]: This whole sequence seems overly complicated for what's essentially three if/then checks.  

Why not something like:
```
    if (readPropertiesFromClassLoader(propsFilename, archProps,
            Thread.currentThread().getContextClassLoader())
    || readPropertiesFromClassLoader(propsFilename, archProps,
            ClassLoader.getSystemClassLoader())
    || readPropertiesFromClassLoader(propsFilename, archProps,
            FileUtil.class.getClassLoader())) {
        return archProps;
    }
```

Comment (ngyukman) [reply oshi-core/src/main/java/oshi/util/FileUtil.java:234]: when I was testing, Thread.currentThread().getContextClassLoader() == ClassLoader.getSystemClassLoader() but perhaps it's early optimization for that

Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/FileUtil.java]: If you'd prefer to reorder them (FileUtil class 2nd and System 3rd?) that's fine with me.

Comment (ngyukman) [reply oshi-core/src/main/java/oshi/util/FileUtil.java]: In most of the application I think all 3 classloaders are the same, so it shouldn't matter too much
`FileUtil.class.getClassLoader()` should always gets the default properties as it is the classloader loading the oshi-core.jar unless the jar is being altered / shaded, keeping it as the last one seems better for others need to override the properties

Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/FileUtil.java]: OK, great!  I'll go ahead and merge this!

**Comment by dbwiddis** [comment]
Heh. I just realized you committed this to 4.X.  Pulling it over to the main branch too.

File: oshi-core/src/main/java/oshi/util/FileUtil.java (hunk 1 above)
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/FileUtil.java]: Almost a year later, I see why you did this and wish I'd left your initial contribution alone. :)

**Pull request merged**